internal/controller/meshfederation_controller_test.go

```diff
@@ -0,0 +1,30 @@
+// Copyright Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the License);
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an AS IS BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	. "github.com/onsi/ginkgo/v2"
```

Comment:
I wonder if we shouldn't consider unifying test framework approach so that we have consistent and familiar code base across different layer of testing.

Reply:
Not part of this PR - general thing to consider.

Reply:
I removed this template.

Reply:
Pity, as I find ginkgo superior for writing tests :P

Reply:
Feel free to use it when you will write tests.

Reply:
I am considering it, though I would sacrifice my own personal preference for consensus and uniformity across the code base. I can demo something next sync however.